### DISCUSSION TITLE
Resolve doesn't use machine-id for DRM

### DISCUSSION THREAD

**Discussion by trafotin** [fat-tire/resolve#45] [General]
Hello,
I have run some extensive tests with DaVinci Resolve Studio and did a write up about it here and in a video essay: [https://trafotin.com/v/davinci-resolve](https://trafotin.com/v/davinci-resolve)

I discovered by accident that Resolve requires a static MAC address in order to stay registered. Using a random MAC address will require you to re-register Resolve every time NetworkManager restarts or reconnects. There's probably a relationship with vendor OUIs so they can limit piracy, but not exactly sure. It's the only reliable reason why they require a static MAC.

As for machine-id, if you change your machine-id to something generic, it has no impact on Resolve at all.

Hope this helps.

**Comment by fat-tire** [comment]
Thanks for the insight/confirmation! I'm using a dongle so was never able to check.

A couple quick thoughts from your write-up:

* If you use the `RESOLVE_BUILD_X264_ENCODER_PLUGIN=Y` flag, you can automatically build the x264 plugin for use on non-NVidia systems (or for otherwise non-supported systems, which I think may include the free version?)  I actually prefer it over the H264 output, it's comparable in speed for me, and I'm actually thinking of making this the default behavior as it's been working now for like 2 years. It git fetches & builds from latest x264 source, so (I think) optimizes for the processor you're actually running on.

* Apparently docker [sets a unique MAC to each container](https://stackoverflow.com/questions/42946453/how-does-the-docker-assign-mac-addresses-to-containers), which can also be set with the `--mac-address` argument to the `docker run` command.  I would think this would require root on podman, but apparently it [works](https://docs.podman.io/en/latest/markdown/podman-network-connect.1.html) there too?  I'll have to try that out.  Regardless, perhaps there should be either a default MAC address set per container or a flag that could be set to override the default. Or is this too niche?

I haven't heard anyone complain about having to constantly re-register with every new `./build.sh`, so I'm guessing that there's some MAC consistency between builds.  If there's any call for it, someone let me know!

Thanks again,
ft

**Comment by fat-tire** [comment]
Oh I just found that there's a "convert to discussion" button, so I'm gonna convert this over if you have any further thoughts.

**Comment by trafotin** [comment]
The MAC address thing is related to the host system. The certificate signing business is restricted to the container. I actually used Distrobox and Fedora, but a friend shared this repo and I had already done the testing.

I'm working with some of the DaVinciBox people to get dGPUs sorted. I believe your script is also affected.
